GVFS/GVFS.Virtualization/Projection/GitIndexProjection.FolderData.cs

```diff
@@ -49,6 +51,26 @@ public void Include()
                 }
             }
 
+            public string HashedFileShas()
+            {
+                byte[] sha = new byte[20];
+                using (HashAlgorithm hash = SHA1.Create())
+                {
+                    for (int i = 0; i < this.ChildEntries.Count; i++)
+                    {
+                        if (!this.ChildEntries[i].IsFolder)
+                        {
+                            FileData fileData = (FileData)this.ChildEntries[i];
+                            fileData.Sha.ToBuffer(sha);
+                            hash.TransformBlock(sha, 0, 20, null, 0);
+                        }

```

Comment:
This needs to be updated to also track any folders so it needs an else statement here.

Since the folder might not have a sha for it yet and we don't want to recurse through the folders to generate the sha, the logic should be to use the sha of the folder if there is one and if there isn't one use the name of the folder since that will be unique for the folder.

Reply:
We want to just hash the **names** of the files/folders in the directory and not the content (so it wouldn't need to recurse)